src/main/java/com/ita/if103java/ims/mapper/UserDtoMapper.java

```diff
@@ -0,0 +1,72 @@
+package com.ita.if103java.ims.mapper;
+
+import com.ita.if103java.ims.dto.UserDto;
+import com.ita.if103java.ims.entity.User;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class UserDtoMapper {
+
+    public User convertUserDtoToUser(UserDto userDto) {
+        if (userDto == null) {
+            return null;
+        } else {
+            User user = new User();
+            user.setId(userDto.getId());
+            user.setFirstName(userDto.getFirstName());
+            user.setLastName(userDto.getLastName());
+            user.setEmail(userDto.getEmail());
+            user.setRole(userDto.getRole());
+            user.setCreatedDate(userDto.getCreatedDate());
+            user.setUpdatedDate(userDto.getUpdatedDate());
+            user.setActive(userDto.isActive());
+            user.setAccountId(userDto.getAccountId());
+            return user;
+        }
+    }
+
+    public UserDto convertUserToUserDto(User user) {
+        if (user == null) {
+            return null;
+        } else {
+            UserDto userDto = new UserDto();
+            userDto.setId(user.getId());
+            userDto.setFirstName(user.getFirstName());
+            userDto.setLastName(user.getLastName());
+            userDto.setEmail(user.getEmail());
+            userDto.setPassword(user.getPassword());
+            userDto.setRole(user.getRole());
+            userDto.setCreatedDate(user.getCreatedDate());
+            userDto.setUpdatedDate(user.getUpdatedDate());
+            userDto.setActive(user.isActive());
+            userDto.setEmailUUID(user.getEmailUUID());
+            userDto.setAccountId(user.getAccountId());
+            return userDto;
+        }
+    }
+
+    public List<User> convertToUserList(List<UserDto> userDtoList) {
+        if (userDtoList == null)
+            return new ArrayList<>();
+        else {
+            List<User> list = new ArrayList<>();
+            userDtoList.forEach(item -> list.add(convertUserDtoToUser(item)));
+            return list;
+        }
```

Comment:
I would recommend to use `Optional` and  `stream().map(...)` for those stuff. I'm not sure about the code, but the main  idea looks like:
```java
public List<User> convertToUserList(List<UserDto> userDtoList) {
      return Optional.ofNullable(userDtoList)
                     .orElse(Colletions.emptyList())
                     .stream()
                     .map(this::convertUserDtoToUser)
                     .collect(Collectors.toList());
}
```